airflow/models/taskinstance.py

```diff
@@ -2715,29 +2727,26 @@ def signal_handler(signum, frame):
 
             # Execute the task
             with set_current_context(context):
-                result = self._execute_task(context, task_orig)
+                result, rendered_map_index = self._execute_task(context, task_orig, jinja_env=jinja_env)
```

Comment:
I wonder if we should just try to render the template _regardless_ of the task at all. Something like

```python
def _render_map_index(...):
    ...

try:
    result = self._execute_task(context, task_orig)
except Exception:
    # If the task failed, swallow rendering error so it doesn't mask the main error.
    with contextlib.suppress(jinja2.TemplateSyntaxError, jinja2.UndefinedError):
        _render_map_index(...)
    raise
else:
    # If the task succeeded, render normally to let rendering error bubble up.
    _render_map_index(...)
```

Reply:
💡 did that and tested it, works as intended the template error is suppressed for failed tasks. (I also like that this version does not involve touching `_execute_task` 😅 )